src/components/AddBook.js

```diff
@@ -1,22 +1,30 @@
 import React, { useState } from 'react';
-import { useSelector, useDispatch } from 'react-redux';
+import { useDispatch } from 'react-redux';
 import { addBook } from '../redux/books/booksSlice';
 import '../styles/AddBook.css';
 import Button from './Button';
 
 function AddBook() {
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
-  const booksArr = useSelector((state) => state.book.books);
+  // const booksArr = useSelector((state) => state.book.books);
   const dispatch = useDispatch();
-  const clickHandler = (e) => {
-    const id = booksArr.length + 1;
+  // useEffect(() => {
+  //   dispatch(fetchBooks());
+  // }, []);
+  // const transformedData = Object.entries(booksArr).map(([id, items]) => {
+  //   const [item] = items; // assume there is only one item in each array
+  //   return { id, ...item };
+  // });
```

Comment:
- [ ] The function to add a new book is not working as expected, please fix the issues. Besides, the removeBook function has issues because I had to click it twice to remove the book. Please 🙏🏽 fix the issues for a better UX. 👇🏽 

https://github.com/KUBAHO3/book-store/assets/58642949/6c9ace6f-e24a-4bba-b306-27dde8652990